lib/preprocessor.py

```diff
@@ -215,8 +215,11 @@ def server_stop(self, ctype, sname, opts):
         if sname not in self.servers:
             raise LuaPreprocessorException(
                 'Can\'t stop nonexistent server {0}'.format(repr(sname)))
-        self.connections[sname].disconnect()
-        self.connections.pop(sname)
+        try:
+            self.connections[sname].disconnect()
+            self.connections.pop(sname)
+        except KeyError:
+            pass
```

Comment:
I found that this situation is triggered by the `replication/election_qsync_stress.test.lua` test, when it attempts to stop an already stopped server.

We should not silently ignore an incorrect command and return to the test like the command was successful. The right behaviour would be:

1. Print a warning on the terminal.
2. Don't stop a test.
3. Raise a Lua error in the calling test code.

I have implemented it in the the first two commits in PR #237 ('Handle incorrect test-run commands gracefully', 'Give meaningful error messages for failed commands').